### ISSUE TITLE
"Broken" provider field in Discord when tweet contains a Video

### ISSUE THREAD

**Issue by ThatRedKite** [FxEmbed/FxEmbed#557]
While developing my Discord i have noticed that a tweet containing a Video will have a broken "provider" field on Discord containing the views, retweets and likes instead of `FixTweet/FixupX`. A few tests have shown that this seems to only affect Tweets with a Video:

### Tweet with an image:
![grafik](https://github.com/FixTweet/FixTweet/assets/48386182/1e174525-0d16-41f1-b499-5a1219099f29)

```json
"author": {
  "name": "1,050 💬    1,270 🔁    20,089 ❤️",
  "url": "https://twitter.com/OlafScholz/status/1466490325879373830"
},
"provider": {
  "name": "FixTweet / FixupX",
  "url": "https://github.com/FixTweet/FixTweet"
},
```

### Tweet with only text:
![grafik](https://github.com/FixTweet/FixTweet/assets/48386182/63709f17-0d0f-4100-9998-d3af66859e6c)

```json
  "author": {
    "name": "5,398 💬    586 🔁    7,083 ❤️",
    "url": "https://twitter.com/OlafScholz/status/1464561866152230917"
  },
  "provider": {
    "name": "FixTweet / FixupX",
    "url": "https://github.com/FixTweet/FixTweet"
  },
```

### Tweet with a video:

![grafik](https://github.com/FixTweet/FixTweet/assets/48386182/d4e247b6-0e7f-4f96-b420-4a08c754e250)

```json
"author": {
  "name": "Everyone can be absolutely sure that Germany stands side by side with its allies and especially the United States when it comes to Russia and the Ukraine. We are prepared to take all necessary steps together. There is no disagreement in this situation.",
  "url": "https://twitter.com/OlafScholz/status/1490966801844674560"
},
"provider": {
  "name": "690 💬  490 🔁  2,921 ❤️",
  "url": "https://twitter.com/OlafScholz/status/1490966801844674560"
},
  ```

Unfortunately this arrangement makes it rather impossible to implement special behavior based on the "name" field because it is suddenly completely different if a tweet contains a video. It also does not look very consistent because suddenly the entire text of a tweet becomes a link.

**Comment by dangeredwolf** [comment]
This appears to because of a lack of system emoji support. What OS are you running?

**Comment by ThatRedKite** [comment]
> This appears to because of a lack of system emoji support. What OS are you running?

No, I think you might have misunderstood me. The issue I am talking about is not the lack of emojis, that part is on purpose, but it's the Emoji block appearing the the wrong part of the Embed if it contains a video and the `FixTweet / FixupX` part disappearing completely.

**Comment by dangeredwolf** [comment]
We are working within the confines of Discord's embed implementation and unfortunately it does not allow text at the same time as video. This is why they look different. We can't do anything about that from a website embed, so the strange appearance is by design so we can actually fit the contents of the tweet in. This limitation doesn't apply to other apps such as Telegram, where videos can be put with text no problem.

**Comment by ThatRedKite** [comment]
> Ah, my bad for the misunderstanding. So for that problem, we are working within the confines of Discord's embed implementation and unfortunately it does not allow text at the same time as video. This is why they look different. We can't do anything about that from a website embed, so the strange appearance is by design so we can actually fit the contents of the tweet in. This limitation doesn't apply to other apps such as Telegram, where videos can be put with text no problem.

Well, unfortunately that implementation leads to the "name" field of the "provider" suddenly becoming completely useless for actually finding out where something is coming from and it's suddenly the likes and comments which is a rather useless piece of information.

**Comment by dangeredwolf** [comment]
All of our embeds already contain the name and username of the person who posted the tweet, so I'm having a difficult time understanding what information you are missing by our implementation of video and non-video statuses?

**Comment by ThatRedKite** [comment]
> All of our embeds already contain the name and username of the person who posted the tweet, so I'm having a difficult time understanding what information you are missing by our implementation of video and non-video statuses?

I am talking about this here in particular

With a video:
```json
"provider": {
  "name": "690 💬  490 🔁  2,921 ❤️",
  ...
 ```
 The "name" of the provider is the likes and comments.

With an image or just text: 
 ```json
   "provider": {
    "name": "FixTweet / FixupX",
    ...
```
 Here the name is the actual name of the provider of the content. Without the provider name being the actual name of the provider of the content makes it really hard to easily implement special behavior for each possible video provider:
 
This really makes it not possible to make a simple switch statement for the name of the "provider" because you will have to implement some kind of special case just for FixTweet and look at the URL of the provider instead (with a bunch of different regexes and whatnot).

**Comment by dangeredwolf** [comment]
I made the assumption that people can see the URL attached to it and therefore can see where it came from. Like I said, since Discord doesn't let you include text in video embeds, we are limited in terms of what fields we can use. When I implemented this recently I did it because I personally think it's more useful to the user to show interaction information than a link to our GitHub. Including this alongside the text of the tweet looks pretty bad imho which is why I haven't done it (and similar reason why interactions counts don't appear in Telegram for instance)

**Comment by dangeredwolf** [comment]
Fixed as it's no longer applicable
